iceaddr/addresses.py

```diff
@@ -51,27 +51,33 @@ def iceaddr_lookup(
     # Look up postcodes for placename if no postcode is provided
     if placename and not postcode:
         pc = postcodes_for_placename(placename.strip())
+    query = "SELECT * FROM stadfong WHERE"
+    name_fields = ["heiti_nf=?", "heiti_tgf=?"]
+    if not number:
+        # Add lookup for churches and places of interest like Harpa
+        name_fields.append("serheiti=?")
+    query += "({})".format(" OR ".join(name_fields))
+    args = [street_name] * len(name_fields)
 
-    q = "SELECT * FROM stadfong WHERE (heiti_nf=? OR heiti_tgf=?)"
-    sqlargs = [street_name, street_name]
```

Comment:
Happy to use `q` instead of `query` and `sqlargs` instead of `args` ... let me know :)